.gitlab-ci.yml

```diff
@@ -19,6 +19,21 @@ cppcheck:
   tags:
     - docker
 
+flake8:
+  stage: build
+  image: pipelinecomponents/flake8:latest
+  variables:
+      FLAKE8_FILES: "tests/capi.py"
+  script:
+    - flake8 --verbose $FLAKE8_FILES
+
+flake8-diff:
+  stage: build
+  image: pipelinecomponents/flake8:latest
+  script:
+    - apk add git
+    - git diff -U0 origin/master | flake8 --verbose --diff
```

Comment:
Can it always safely run on partial code taken out of their context?

Reply:
It actually runs on the full file, it just only reports errors in the hunks mentioned by the diff.

Reply:
That's awesome! :-)